analytic-ui-generator/src/main/java/com.wipro.ats.bdre.augen/AnalyticUIGeneratorMain.java

```diff
@@ -0,0 +1,111 @@
+import com.wipro.ats.bdre.md.dao.jpa.Process;
+import org.apache.commons.cli.CommandLine;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+
+/**
+ * Created by cloudera on 4/15/16.
+ */
+public class AnalyticUIGeneratorMain extends MetadataAPIBase {
+    @Autowired
+    AnalyticsAppsDAO analyticsAppsDAO;
+    @Autowired
+    ProcessDAO processDAO;
+
+
+    private static final Logger LOGGER = Logger.getLogger(AnalyticUIGeneratorMain.class);
+    private static final String[][] PARAMS_STRUCTURE = {
+            {"p", "parent-process-id", "Process Id of the process to begin"},
+            {"u", "username", "Username"}
+
+    };
+    public AnalyticUIGeneratorMain() {
+        AutowireCapableBeanFactory acbFactory = getAutowireCapableBeanFactory();
+        acbFactory.autowireBean(this);
+    }
+
+    public static void main(String[] args)  {
+        AnalyticUIGeneratorMain analyticUIGeneratorMain = new AnalyticUIGeneratorMain();
+        analyticUIGeneratorMain.table2DAOAnalyticUI(args);
+
+
+    }
+    @Override
+    public Object execute(String[] param){
+        return null;
+    }
+
+    public void table2DAOAnalyticUI(String[] args){
+        CommandLine commandLine = new AnalyticUIGeneratorMain().getCommandLine(args, PARAMS_STRUCTURE);
+        String pid = commandLine.getOptionValue("parent-process-id");
+        LOGGER.debug("processId is " + pid);
+        String username = commandLine.getOptionValue("username");
+        LOGGER.debug("username is " + username);
+        processDAO.securityCheck(Integer.parseInt(pid),username, "execute");
+        String outputFile = "";
+
+
+        //Getting sub-process for process-id
+        List<ProcessInfo> processInfos = new GetProcess().execute(new String[]{"--parent-process-id", pid , "--username" , username});
+        // Getting properties related with flume action for every process
+        StringBuilder addFlumeProperties = new StringBuilder();
+        com.wipro.ats.bdre.md.dao.jpa.AnalyticsApps analyticsApps = new com.wipro.ats.bdre.md.dao.jpa.AnalyticsApps();
+
+        for (ProcessInfo processInfo : processInfos) {
+            if (processInfo.getParentProcessId() == 0) {
+                Process process = new Process();
+                process.setProcessId(processInfo.getProcessId());
+                analyticsApps.setProcess(process);
+                continue;
+            }
+            GetProperties getProperties = new GetProperties();
+            java.util.Properties flumeProperties = getProperties.getProperties(processInfo.getProcessId().toString(), "analytics-app");
+            Enumeration e = flumeProperties.propertyNames();
+
+            if (!flumeProperties.isEmpty()) {
+                while (e.hasMoreElements()) {
+                    String key = (String) e.nextElement();
+                    if ("app-desc".equals(key)){
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)